### ISSUE TITLE
Add files to DB if metadata parsing fails

### ISSUE THREAD

**Issue by IrosTheBeggar** [IrosTheBeggar/mStream#112]
Right now mStream will skip the file completely if metadata parsing fails.   There should still be an entry that contains the sha256 hash and other basic info

**Comment by dkanada** [comment]
I am guessing this is from my bug haha. Technically there could be a fallback method that uses the folder structure to get some other information as well. Most people use some variation of albumartist/album/song.flac for their music files, and mStream could specify a library format similar to [kodi](https://kodi.wiki/view/Naming_video_files/TV_shows) for this to function properly.

**Comment by dkanada** [comment]
Just saw the project was previously using sqlite, why did you switch to loki? I was going to suggest moving to a relational database for less ambiguity. Wanted to add things like artist cover art or vpath settings.

**Comment by IrosTheBeggar** [comment]
@dkanada good question about loki vs sqlite

The goal of mStream is to be the easiest music server to setup and deploy.  In order to be the easiest mStream cannot have any external dependencies.  This rules out all databases that run as a separate server like MySQL/MongoDB/etc. 

This 'no external dependenices' rule makes things a bit tricky. It means mStream's DB needs to be written in javascript, [or needs to written in a language that can be compiled as NodeJS addon](https://nodeaddons.com/building-an-asynchronous-c-addon-for-node-js-using-nan/).  SQLite had a NodeJS port called node-sqlite3 that mStream originally used.

There's two ways to use a NodeJS addon.  Either compile it and distribute it with mStream, or let the user compile it on installation.  I tried compiling and distributing it, but I got a lot of complaints from Windows users about the DB throwing mysterious errors or just straight failing it initialize.  There was the fallback solution of compiling SQLite on the users computer, but that meant the user needed to have some additional dependencies like, node-gyp, python, and GCC. 

Additionally the node-sqlite3 library had a lot of bloat, which caused the mStream Express installation to take up to 15 min on slower systems.

So I replaced SQLite with LokiJS. Loki has the advantage of being a javascript library, so it just worked with no additional dependencies. I stopped getting complaints about DB errors after the switch, so I would say it was a success.  

In the future it might be possible to compile SQLite directly to webassembly as a way to solve this problem, but I'm going to wait a bit before looking into that
